src/genie/libs/parser/iosxr/show_mpls.py

```diff
@@ -71,6 +71,291 @@ def cli (self, output=None):
             
         return mpls_dict
 
+# ==============================================
+# Parser for 'show mpls ldp discovery'
+# ==============================================
+class ShowMplsLdpDiscoverySchema(MetaParser):
+
+    """Schema for show mpls ldp discovery
+                  show mpls ldp discovery detail
+                  show mpls ldp afi-all discovery
+                  show mpls ldp discovery <ldp>
+                  show mpls ldp vrf <vrf> discovery
+                  show mpls ldp vrf <vrf> discovery detail
+    """
+
+    schema = {
+        'vrf': {
+            Any(): {
+                Optional('local_ldp_identifier'): {
+                    Any(): {
+                        Optional('discovery_sources'): {
+                            'interfaces': {
+                                Any(): {
+                                    Optional('source_ip_addr'): str,
+                                    Optional('transport_ip_addr'): str,
+                                    Optional('xmit'): bool,
+                                    Optional('recv'): bool,
+                                    Optional('hello_interval_ms'): int,
+                                    Optional('hello_due_time_ms'): int,
+                                    Optional('quick_start'): str,
+                                    Any(): {
```

Comment:
this is always `ldp_id`, right? if yes, let's have it statically.

Reply:
I'm not sure, I will have to check the Yang.

Reply:
According to https://github.com/YangModels/yang/blob/master/vendor/cisco/xr/653/Cisco-IOS-XR-mpls-ldp-oper-sub1.yang#L3234 which is used here https://github.com/YangModels/yang/blob/master/vendor/cisco/xr/653/Cisco-IOS-XR-mpls-ldp-oper.yang#L89 it could take ldp-id, label-space-id or lsr-id.

Reply:
```
    Any(): {
        Any(): {
```
what I meant was, this is widely opened. Let's have like below:
```
    'ldp_id': {
        any(): {
```
based on current implementation, we will have only `ldp_id`, I guess.
If anything else needs to be added, we can add/update schema at that time.

Reply:
Ok I see, I've patched it.

Reply:
confirmed